pkg/service/oidc4vc/oidc4vc_service.go

```diff
@@ -232,11 +262,51 @@ func (s *Service) PrepareCredential(
 		return nil, fmt.Errorf("claim endpoint returned status code %d", resp.StatusCode)
 	}
 
-	// TODO: Prepare and sign credential
+	var claimData map[string]interface{}
+
+	if err = json.NewDecoder(resp.Body).Decode(&claimData); err != nil {
+		return nil, fmt.Errorf("decode claim data: %w", err)
+	}
+
+	// prepare and sign credential
+	vc := &verifiable.Credential{
+		Context: tx.CredentialTemplate.Contexts,
+		ID:      uuid.New().URN(),
+		Types:   []string{"VerifiableCredential", tx.CredentialTemplate.Type},
+		Issuer:  verifiable.Issuer{ID: tx.CredentialTemplate.Issuer},
+		Subject: verifiable.Subject{
+			ID:           req.DID,
+			CustomFields: claimData,
+		},
+		Issued: util.NewTime(time.Now()),
+	}
+
+	var credential interface{}
+
+	switch tx.CredentialFormat {
+	case vcsverifiable.Jwt:
+		claims, jwtClaimsErr := vc.JWTClaims(false)
+		if jwtClaimsErr != nil {
+			return nil, fmt.Errorf("create jwt claims: %w", jwtClaimsErr)
+		}
+
+		credential, err = claims.MarshalUnsecuredJWT()
```

Comment:
Dont we need to call `claims.MarshalJWS` here?

Reply:
see above